### ISSUE TITLE
Move away from using project options

### ISSUE THREAD

**Issue by wusatosi** [bemanproject/execution26#101]
In weekly sync of 12-02.
We decide to discourage use of third-party CMake build dependency, specifically project_options.
We probably need to move away from this in execution26.

This is low priority.

**Comment by ClausKlein** [comment]
The arguments are weak.

You have no concept and no knowledge to create this `cmake infrastructure` yourself.
`FetchContent` is used in other projects to get **GMock**, so offline build is no really a good argument against project_options.

Finally, you have no usable presets and no real ideas how to export `cmake config packages` in a usable way.

Bad Luck

**Comment by wusatosi** [comment]
> The arguments are weak.
> 
> You have no concept and no knowledge to create this `cmake infrastructure` yourself. `FetchContent` is used in other projects to get **GMock**, so offline build is no really a good argument against project_options.
> 
> Finally, you have no usable presets and no real ideas how to export `cmake config packages` in a usable way.
> 
> Bad Luck

Hey, I pitched this to the meeting on a neutral stance.
I didn't make any arguments against this.
I am simply flagging this here and noting the consensus.

**Comment by ClausKlein** [comment]
@wusatosi sorry, I was meaning the `beman project`, not you personally !

**Comment by dietmarkuehl** [comment]
I don't think we had an exhaustive discussion that on Monday. I think we should assess what we want to achieve and what is an effective way to achieve things. Removing useful support without a strong reason seem premature. I don't think this repo is actually the place to discuss that, though: I imagine this discussion to be held on [`examplar`](https://bemanproject/examplar) and other repos following suit eventually.

**Comment by steve-downey** [comment]
I discovered last night that the [project options](https://github.com/aminya/project_options/) , at least how we're using it, overrides the configuration sets to " Debug Release RelWithDebInfo MinSizeRel " even when beman.execution is being integrated via FetchContent. So multi-config for Asan, Gcov, etc becomes broken. 

I'll have a PR to remove soon-ish.
